### PR TITLE
Add support for `query/saveMarkdown` request

### PR DESCRIPTION
**Pull request by benrr101**
**Description**: This PR adds the necessary code to wire up the `query/saveMarkdown` request that is proposed in microsoft/sqltoolsservice#1705. This change will be used by [insert ADS PR here] to connect between sqltoolsservice and ADS.

**Testing**:
To test:
* `yarn compile` on sqlops-dataprotocolclient
* Pull down microsoft/azuredatastudio#[insertADSPRHere]
* In ADS, modify extensions/mssql/package.json to point to this local build of dataprotocolclient
* `yarn compile` on ADS
* Pull down microsoft/sqltoolsservice#1705
* Build solution on sqltoolsservice
* Copy sqltoolsservice bin folder to extensions/mssql/sqltoolsservice drop
* Launch ADS
* Run a query that selects something
* Observe new button on grid for exporting to Markdown
* Observe new right click action to export to Markdown on the grid
* Try it! Observe that it outputs a markdown file

### CODE REVIEW

**Review by Charles-Gagnon — COMMENTED** [review]
Also update package.json version (1.2.4 -> 1.3.0)

**Comment by benrr101** [comment]
> Also update package.json version (1.2.4 -> 1.3.0)

Done 👍

File: src/protocol.ts [hunk 1]
```diff
@@ -413,6 +413,13 @@ export namespace SaveResultsAsJsonRequest {
 }
 // --------------------------------- </ Save Results as JSON Request > ------------------------------------------
 
+// --------------------------------- < Save Results as Markdown Request > ------------------------------------------
+// save results in Markdown format
+export namespace SaveResultsAsMarkdownRequest {
+	export const type = new RequestType<azdata.SaveResultsRequestParams, azdata.SaveResultRequestResult, void, void>('query/saveMarkdown');
+}
+// --------------------------------- </ Save Results as JSON Request > ------------------------------------------
```
Comment (Charles-Gagnon) [inline src/protocol.ts:421]: `as Markdown Request`

**Review by Charles-Gagnon — APPROVED** [review]

File: src/protocol.ts (hunk 1 above)
Comment (benrr101) [reply src/protocol.ts:421]: Done, I think :)

**Comment by Charles-Gagnon** [comment]
I'll get a PR out for updating this in the ADS repo since there's a number of extensions that all should be bumped up to the latest.

**Pull request merged**